### DISCUSSION TITLE
quadcopter example

### DISCUSSION THREAD

**Discussion by soldierofhell** [osqp/osqp#558] [Mathematical Modeling]
Hi,
Could you provide description or a reference to quadcopter model. It seems like controls are rotors velocities, but hard to guess what are the states and what is the nonlinear model, quadrotor parameters, etc.
Regards,

**Comment by goulart-paul** [comment]
It is the quadcopter model defined in [this](https://github.com/ferreau/mpcBenchmarking/) benchmark collection.  The relevant paper for that benchmark collection is [here](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=7330931).

They don't seem to say in the benchmarking paper where the data comes from though.   Comments within the repo itself ([here](https://github.com/ferreau/mpcBenchmarking/blob/5c1e104973a6e19cfc8384665a7eb69f52ee74f7/benchmarks/Benchmark_quadcopter.m#L36)) say that it came from the MPT3 toolbox, probably [this](https://www.mpt3.org/Main/BenchmarkExamples), by way of IMT Lucca.

Making a guess now, but the original source of the model was perhaps [this paper](http://cse.lab.imtlucca.it/~bemporad/publications/papers/adhs09-quadcopter.pdf).
